tests/test_traced_data.py

```diff
@@ -0,0 +1,301 @@
+import collections
+import time
+import unittest
+import six
+
+from core_data_modules.traced_data import TracedData, Metadata
+
+
+class TestTracedData(unittest.TestCase):
```

Comment:
I think it would be a good idea to add a 'demo_test' method that looked like we were expecting the client code to look like as well as these methods. That way the tests can become part of the explanatory mechanism for the data structure.

Reply:
Wouldn't doctest be a better place for this?

Reply:
Added a doctest